team-agreements/code-with-agreements.md

```diff
@@ -0,0 +1,40 @@
+# Code With Agreements
+
+[CSE](../CSE.md) code with **CP (the Customer or Partner)**.
+
+Both teams from different companies gather as one team, with different backgrounds.
+This brings diversity and that's the point of the engagement, but the path may be bumpy, especially if rules are blurry or inexistent.
+It's also important to prepare for the end of the engagement, when code will still have to be maintained by CP, and/or open sourced by CSE or CP.
+
+For all these reasons, both parties need to agree on a few principles.
+
+## Tech Lead Role
+
+The Tech Lead role can be held by one person from CSE, or one person from CSE and one person from CP.
+
+When the role is played by both parties, they must agree on their respective scopes.
+For instance, if CP is an automotive company and the project is on Azure,
+the CSE Tech Lead will lead on Azure (ex: architecture, security, services choices etc.), and the CP Tech Lead will lead on everything related to automotive (ex: specific software, functional code, security and architectures decisions to follow standards/regulations etc.)
+
+If CP already have strong habits on software development, having the duo will be very beneficial to make sure
+these habits are both respected and challenged (if needed).
+
+## Product Owner Role
+
+The **PO (Product Owner)** role can also be held by one person from CSE, or one person from CSE and one person from CP. But in most cases, it's better to have a PO on both side.
+
+When the role is played by a duo, both people will work closely together to maintain a unique set of priorities for the backlog.
+When it comes to backlog refinement, the CP PO may have a stronger focus on what needs to be done while the CSE PO may have more ideas on if and how that can be done.
+
+This is a suggestion, and it must be refined and agreed by both PO role players, based on the context of the engagement.
+
+## Other roles
+
+The team also need to agree on how important it is to have people from CSE or CP in key activities.
+Here are a few questions that should be answered as part of the agreement:
+
+- Do we need to have people from CSE and/or CP for
+  - design reviews?
+  - code reviews?
+  - pair programming?
+- After a first sprint where one person of CSE starts playing the role of Scrum Master, do we rotate and switch between a CSE and a CP Scrum Master? NB: we strongly encourage to rotate and switch between a CSE and a CP Scrum Master. But we recommend to have only one Scrum Master per sprint to avoid confusion in the team.
```

Comment:
Note: there is another PR in the works where they are replacing the title Scrum Master - would be good to sync these two PRs

Reply:
I agree that the term may have to be changed, but that should be done while the second PR arrives in master. Either a consensus was found for the replacing term, it's in the master (or main) branch and this PR will have to change, or this PR goes to master (or main) and the PR will take the new text into account and rename in the new page as well.

Reply:
The other PR is merged now - with scrum master replaced by the term Process Lead

Reply:
OK. I updated this one. Thanks